src/main/java/org/apache/maven/plugins/shade/resource/ManifestResourceTransformer.java

```diff
@@ -46,6 +46,8 @@
     // Configuration
     private String mainClass;
 
+    private boolean skipRelocators;
```

Comment:
Do we really need this skip parameter? If the manifest file should be transformed and there are relocated classes, just apply them. I see no reason to pollute this class with a skip parameter.

Reply:
Agree but the algorithm being a bit fragile until we use a full osgi header parsers/rewriter with exception algorithms for the general case, I'd prefer a parameter to disable it for not osgi case. I can enhance the parsing but thought it is not the core of shade plugin so is kot worse that logic. If wrong, happy to revise this.
Not perfect state but secured - would be sad to break a project for that new behavior IMHO.
Wdyt?

Reply:
Is there a way to discover if it is a OSGi project, e.g. via the packaging?

Reply:
Yes and no, they will have Bundle-Version entry in the manifest but most of the modules will be potential bundles but more often used in se/ee. For ex, the test i wrote used an openwebbeans output. It is used in se context, tomee, meecrowave (all not osgi) but also aries-cdi (osgi) or as plain osgi bundle.
This is why i introduced this boolean.

Reply:
Looking at the code, it should be possible with a dedicated relocator instead of the SimpleRelocator. In this case you'll need a more delicate way than the brute search and replace all.

Reply:
Hmm, sounds like an option but current code works well for jakarta relocation too so I'm mixed to make the conf more complex/verbose for end user.
Is having the boolean AND dedicated relocators ok for you? Boolean could be reversed and named "inheritDefaultRelocators".
Wdyt?

Reply:
FYI, I updated the code to reflect the proposal

Reply:
Once a skip parameter is introduced, it is very hard to get rid of it, so I'll try to block it when possible(it is not a common practice as many users use as argument, it is just a lazy hack). The enduser shouldn't notice the difference. I'm also working on the maven-shade-plugin, so I'll give it a try as well.